docs/modules/ROOT/pages/governance.adoc

```diff
@@ -0,0 +1,192 @@
+:accesscontrol-component: xref:api/access.adoc#AccessControlComponent[AccessControlComponent]
+:src5-component: xref:api/introspection.adoc#SRC5Component[SRC5Component]
+
+== Timelock Controller
+
+The Timelock Controller provides a means of enforcing time delays on the execution of transactions. This is considered good practice regarding governance systems because it allows users the opportunity to exit the system if they disagree with a decision before it is executed.
+
+NOTE: The Timelock contract itself executes transactions, not the user. The Timelock should, therefore, hold associated funds, ownership, and access control roles.
+
+=== Operation lifecycle
+
+Timelocked operations are identified by a unique id (their hash) and follow a specific `OperationState` lifecycle:
+
+`Unset` → `Waiting` → `Ready` → `Done`
+
+=== Timelock flow
+
+==== Schedule
+
+:schedule: xref:api/governance.adoc#ITimelock-schedule[schedule]
+:get_timestamp: xref:api/governance.adoc#ITimelock-get_timestamp[get_timestamp]
+
+When a proposer calls {schedule}, the `OperationState` moves from `Unset` to `Waiting`.
+This starts a timer that must be greater than or equal to the minimum delay.
+The timer expires at a timestamp accessible through {get_timestamp}.
+Once the timer expires, the `OperationState` automatically moves to the `Ready` state.
+At this point, it can be executed.
+
+==== Execute
+
+:execute: xref:api/governance.adoc#ITimelock-execute[execute]
+
+By calling {execute}, an executor triggers the operation's underlying transactions and moves it to the `Done` state. If the operation has a predecessor, the predecessor's operation must be in the `Done` state for this transaction to succeed.
+
+==== Cancel
+
+:cancel: xref:api/governance.adoc#ITimelock-cancel[cancel]
+
+The {cancel} function allows cancellers to cancel any pending operations.
+This resets the operation to the `Unset` state.
+It is therefore possible for a proposer to re-schedule an operation that has been cancelled.
+In this case, the timer restarts when the operation is re-scheduled.
+
+=== Roles
+
+{timelock-component} leverages an {accesscontrol-component} setup that we need to understand in order to set up roles.
+
+- `PROPOSER_ROLE` - in charge of queueing operations.
+
+- `CANCELLER_ROLE` - may cancel scheduled operations.
+During initialization, accounts granted with `PROPOSER_ROLE` will also be granted `CANCELLER_ROLE`.
+Therefore, the initial proposers may also cancel operations after they are scheduled.
+
+- `EXECUTOR_ROLE` - in charge of executing already available operations.
+
+- `DEFAULT_ADMIN_ROLE` - can grant and revoke the three previous roles.
+This is a sensitive role that will be granted automatically to the timelock itself and optionally to a second account.
+The latter case may be required to ease a contract's initial configuration; however, this role should promptly be renounced.
```

Comment:
I think moving this into a CAUTION note would be helpful, since it's going to be easier to find.

Reply:
I put lines 60 and 61 as a CAUTION note and kept it here to not stack CAUTION and WARNING notes on the bottom. Let me know what you think

Reply:
I like it, just an extra small suggestion, the warning below looks more like a caution block too.